src/Database/Query.php

```diff
@@ -1412,11 +1413,17 @@ public function values($data)
      *
      * Can be combined with set() and where() methods to create update queries.
      *
-     * @param string $table The table you want to update.
+     * @param string|\Cake\Database\ExpressionInterface $table The table you want to update.
      * @return $this
      */
     public function update($table)
     {
+        if (!is_string($table) && !($table instanceof ExpressionInterface)) {
+            $text = 'Table must be of type string or "%s", got "%s"';
+            $message = sprintf($text, ExpressionInterface::class, gettype($table));
+            throw new InvalidArgumentException($message);
```

Comment:
Please look at other places in the core that throw this exception. To be consistent, it shouldn't be split across 3 lines, and we don't inject class names into the error. Also, `gettype($table)` adds no useful information to the developer. In most cases it will simple output the word "object". Please remove references to the expected types from the error message. This creates multiple spots in the function that have to be updated when the argument type is changed (likely not to happen here).

My only concern with this would be unexpected cases with other DB drivers. There are edge cases where `$table` can be an array.

It's a problematic part of the code. I'm not sure this should be a blocking check. It would be better to allow `mixed` and throw exceptions at the SQL driver level.